contrib/signet/miner

```diff
@@ -501,6 +521,8 @@ def main():
     cmds = parser.add_subparsers(help="sub-commands")
     genpsbt = cmds.add_parser("genpsbt", help="Generate a block PSBT for signing")
     genpsbt.set_defaults(fn=do_genpsbt)
+    genpsbt.add_argument("--poolnum", default=None, type=int, help="Identify blocks that you mine")
+    genpsbt.add_argument("--poolid", default=None, type=str, help="Identify blocks that you mine (eg: /signet:1/)")
```

Comment:
In 338a266a9a08e47bc6dd02175c8fa649f701515d "signet/miner: add support for a poolnum/poolid tag in mined blocks"

Using a mutually exclusive group here and below will let the argument parser enforce that these are mutually exclusive rather than having to enforce it in `get_poolid`.

```suggestion
    poolid_group = genpsbt.add_mutually_exclusive_group()
    poolid_group.add_argument("--poolnum", default=None, type=int, help="Identify blocks that you mine")
    poolid_group.add_argument("--poolid", default=None, type=str, help="Identify blocks that you mine (eg: /signet:1/)")
```

Reply:
Added a commit to use mutually exclusive groups here and in a few other places.